src/stresster.rs

```diff
@@ -0,0 +1,172 @@
+        let mut client = Client::builder()
+            .default_headers(headers.clone())
+            .build()
+            .unwrap();
+
+        if !ssl_cert.is_empty() {
+            let mut buf = Vec::new();
+            fs::File::open(ssl_cert)
+                .unwrap_or_else(|_| panic!("ERROR: Error reading {:?}", ssl_cert))
+                .read_to_end(&mut buf)
+                .expect("ERROR: Error reading file");
+            let cert = reqwest::Certificate::from_pem(&buf).unwrap();
+            client = Client::builder()
+                .add_root_certificate(cert)
+                .default_headers(headers)
+                .build()
+                .unwrap();
+        }
+        match method {
+            HttpMethods::Get => {
+                result = client.get(&*target_url).json(payload).send().await;
+            }
+            HttpMethods::Post => {
+                result = client.post(&*target_url).json(payload).send().await;
+            }
+            HttpMethods::Put => {
+                result = client.put(&*target_url).json(payload).send().await;
+            }
+            HttpMethods::Delete => {
+                result = client.delete(&*target_url).json(payload).send().await;
+            }
+            HttpMethods::Patch => {
+                result = client.patch(&*target_url).json(payload).send().await;
+            }
+        };
+        match result {
+            Ok(r) => {
+                let command = Command::Increment(r.status().as_u16());
+                info!(logger, "Result status code: {}", r.status().as_u16());
+                sender.send(command).await.unwrap();
+            }
+            Err(e) => {
+                let command = Command::Increment(0);
+                error!(logger, "Result error : {}", e);
+                sender.send(command).await.unwrap();
+            }
+        }
+    }
+
+    pub async fn run(&self) {
+        // Generate command line args
+        let matches = get_cmd_args().await;
+
+        // Extract user supplied values
+        let (output_format, config_filename, total_requests) =
+            extract_values_from_args(matches).await;
+
+        let shared_data: Data; // Data shared between tasks
+
+        // Create RequestData from data file
+        let request_data = get_request_data_from_file(config_filename).await;
+        shared_data = Arc::new(request_data);
+
+        // Variables shared between tasks
+        let counter = Arc::new(Mutex::new(HashMap::new())); // Map of Atomic counters to keep HTTP status code count
+        let (sender, receiver) = mpsc::channel(50);
+
+        // Create a logger instance
+        let logger = get_logger(self.log_path.to_owned()).await;
+        let shared_logger = Arc::new(logger);
+
+        // Start counter function
+        let counter_clone = counter.clone();
+        let counting_machine_handle =
+            tokio::spawn(async move { self.counting_machine(counter_clone, receiver).await });
```

Comment:
```suggestion
        let task = self.counting_machine(counter_clone, receiver);
        let counting_machine_handle = tokio::spawn(task);
```
:point_up: Here you are putting a task in a task.

This will probably not solve your issue as this task still depend on `&self` which is a reference and you want the task to outlive the current future (which you can't guarantee).